File: app/src/main/java/com/tsits/tsits_webrtc/fragment/MessageFragment.java
package com.tsits.tsits_webrtc.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.tsits.tsits_webrtc.R;
import com.tsits.tsits_webrtc.activity.MessageTalkingRoomActivity;
import com.tsits.tsits_webrtc.adapter.MessageContantRvAdapter;
import com.tsits.tsits_webrtc.entity.MessageContant;
import com.tsits.tsits_webrtc.inter_face.OnItemClickListener;

import java.util.ArrayList;
import java.util.List;

/**
 * @author YUAN
 * @description:
 * @date :2021/9/15 15:58
 */
public class MessageFragment extends Fragment implements OnItemClickListener {
    private static final String TAG = "MessageFragment";
    ArrayList dataList = new ArrayList<Object>();
    private RecyclerView recyclerView_message;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_message, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        recyclerView_message = getActivity().findViewById(R.id.recyclerView_message);
        initRecyclerView();
    }

    public int getCount() {
        return dataList.size();
    }

    public Object getItem(int position) {
        return dataList.get(position);
    }

    public long getItemId(int position) {
        return position;
    }

    private void initRecyclerView() {
        List dataList = getData();
        MessageContantRvAdapter adapter = new MessageContantRvAdapter(getContext(),this);
        recyclerView_message.setLayoutManager(new LinearLayoutManager(getActivity(), RecyclerView.VERTICAL, false));
        DividerItemDecoration divider = new DividerItemDecoration(getActivity(), DividerItemDecoration.VERTICAL);
        recyclerView_message.addItemDecoration(divider);
        recyclerView_message.setAdapter(adapter);
        adapter.setData(dataList);
    }

    /*
     * 取得列表
     * */
    private List<Object> getData() {
        for (int index = 0; index < 9; index++) {
            dataList.add(
                    new MessageContant(
                            "https://img1.baidu.com/it/u=1502334019,908245671&fm=26&fmt=auto&gp=0.jpg",
                            "group:" + index,
                            "time:" + index
                    )
            );
            Log.d("message",
                    "---------------->initRecyclerView datalist size111111:" + dataList.size());
        }
        return dataList;
    }




    @Override
    public void OnItemClick(@NonNull View view, int position) {
        Toast.makeText(getActivity(), "YOU Click is " + view + position, Toast.LENGTH_SHORT).show();
        turnToMessageTalkingRoomActivity(position);
    }

    /*
     * 携带bundle跳转到MessageTalkingRoom
     * */
    private void turnToMessageTalkingRoomActivity(int position) {
        Intent intent = new Intent(getContext(), MessageTalkingRoomActivity.class);
        Bundle bundle = new Bundle();
        bundle.putString("getItem", getItem(position).toString());
//        bundle.putString("getItem", dataList.get(position).toString())
        intent.putExtras(bundle);
        startActivityForResult(intent, 1002);
        Log.d(TAG, bundle.toString());
    }




}

File: app/src/main/java/com/tsits/tsits_webrtc/fragment/MessageFragment.kt
package com.tsits.tsits_webrtc.fragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.tsits.tsits_webrtc.R
import com.tsits.tsits_webrtc.activity.MainActivity
import com.tsits.tsits_webrtc.adapter.MessageContantRvAdapter
import com.tsits.tsits_webrtc.entity.MessageContant
import kotlinx.android.synthetic.main.fragment_message.*


class MessageFragment : Fragment(){


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View? {
        return inflater.inflate(R.layout.fragment_message, null)
    }

    @SuppressLint("SetTextI18n")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
    }

    @SuppressLint("ResourceType")
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initRecyclerView()
        turnToMessageTalkingRoomFragment()

    }



    private fun initRecyclerView() {
        var dataList = getData()
        Log.d("message", "---------------->initRecyclerView datalist size:${dataList?.size}")
        val adapter = MessageContantRvAdapter(context!!)
        recyclerView.layoutManager = LinearLayoutManager(activity, RecyclerView.VERTICAL, false)
        val divider = DividerItemDecoration(activity, DividerItemDecoration.VERTICAL)
        recyclerView.addItemDecoration(divider)
        recyclerView.adapter = adapter
        adapter.setData(dataList)
    }

    private fun getData(): List<Any> {
        val dataList = ArrayList<Any>()
        for (index in 0..9) {
            dataList.add(
                MessageContant(
                    "https://img1.baidu.com/it/u=1502334019,908245671&fm=26&fmt=auto&gp=0.jpg",
                    "group:$index",
                    "time:$index"
                )
            )
            Log.d("message",
                "---------------->initRecyclerView datalist size111111:${dataList?.size}")
        }
        return dataList
    }

    fun turnToMessageTalkingRoomFragment() {
        toolbar_navigation_message.setOnClickListener() {
            val ft = activity!!.supportFragmentManager.beginTransaction()
            ft.replace(R.id.linearLayout4, MessageTalkingRoomFragment())
            ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
            ft.commit()
        }
    }




}